### PR TITLE
Adding child objects for cloud external network parent.

### PR DESCRIPTION
**Pull request by anvitha-jain**

### CODE REVIEW

File: models/cloudtemplate_ip_sec_tunnel.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+const (
+	DncloudtemplateIpSecTunnel        = "uni/tn-%s/infranetwork-%s/extnetwork-%s/vpnnetwork-%s/ipsec-[%s]"
+	RncloudtemplateIpSecTunnel        = "ipsec-[%s]"
+	ParentDncloudtemplateIpSecTunnel  = "uni/tn-%s/infranetwork-%s/extnetwork-%s/vpnnetwork-%s"
+	CloudtemplateipsectunnelClassName = "cloudtemplateIpSecTunnel"
+)
+
+type CloudTemplateforIpSectunnel struct {
+	BaseAttributes
+	NameAliasAttribute
```
Comment (lhercot) [inline models/cloudtemplate_ip_sec_tunnel.go:19]: This class does not have nameAlias

File: models/cloudtemplate_bgp_ipv4.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+const (
+	DncloudtemplateBgpIpv4        = "uni/tn-%s/infranetwork-%s/extnetwork-%s/vpnnetwork-%s/ipsec-[%s]/bgpipv4-[%s]"
+	RncloudtemplateBgpIpv4        = "bgpipv4-[%s]"
+	ParentDncloudtemplateBgpIpv4  = "uni/tn-%s/infranetwork-%s/extnetwork-%s/vpnnetwork-%s/ipsec-[%s]"
+	Cloudtemplatebgpipv4ClassName = "cloudtemplateBgpIpv4"
+)
+
+type BGPIPv4Peer struct {
+	BaseAttributes
+	NameAliasAttribute
```
Comment (lhercot) [inline models/cloudtemplate_bgp_ipv4.go:19]: This class does not have nameAlias

File: models/cloudtemplate_bgp_ipv4.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	BGPIPv4PeerAttributes
+}
+
+type BGPIPv4PeerAttributes struct {
+	Annotation string `json:",omitempty"`
+	Peeraddr   string `json:",omitempty"`
```
Comment (lhercot) [inline models/cloudtemplate_bgp_ipv4.go:25]: Isn't there an ASN attribute in this class?

File: models/cloudtemplate_bgp_ipv4.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+)
+
+const (
+	DncloudtemplateBgpIpv4        = "uni/tn-%s/infranetwork-%s/extnetwork-%s/vpnnetwork-%s/ipsec-[%s]/bgpipv4-[%s]"
+	RncloudtemplateBgpIpv4        = "bgpipv4-[%s]"
+	ParentDncloudtemplateBgpIpv4  = "uni/tn-%s/infranetwork-%s/extnetwork-%s/vpnnetwork-%s/ipsec-[%s]"
+	Cloudtemplatebgpipv4ClassName = "cloudtemplateBgpIpv4"
+)
+
+type BGPIPv4Peer struct {
```
Comment (lhercot) [inline models/cloudtemplate_bgp_ipv4.go:17]: Would add CloudTemplateBGPIPv4Peer to differentiate. Same for other struct and name below.

File: client/cloudtemplateVpnNetwork_service.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/ciscoecosystem/aci-go-client/v2/models"
+)
+
+func (sm *ServiceManager) CreateTemplateforVPNNetwork(name string, template_for_external_network string, infra_network_template string, tenant string, nameAlias string, cloudtemplateVpnNetworkAttr models.TemplateforVPNNetworkAttributes) (*models.TemplateforVPNNetwork, error) {
```
Comment (lhercot) [inline client/cloudtemplateVpnNetwork_service.go:9]: CreateCloudTemplate...
Same for other functions in this file.

File: models/cloudtemplate_vpn_network.go [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+)
+
+const (
+	DncloudtemplateVpnNetwork        = "uni/tn-%s/infranetwork-%s/extnetwork-%s/vpnnetwork-%s"
+	RncloudtemplateVpnNetwork        = "vpnnetwork-%s"
+	ParentDncloudtemplateVpnNetwork  = "uni/tn-%s/infranetwork-%s/extnetwork-%s"
+	CloudtemplatevpnnetworkClassName = "cloudtemplateVpnNetwork"
+)
+
+type TemplateforVPNNetwork struct {
```
Comment (lhercot) [inline models/cloudtemplate_vpn_network.go:17]: CloudTemplate...
Same for other functions in this file.

**Review by lhercot — CHANGES_REQUESTED** [review]

File: models/cloudtemplate_bgp_ipv4.go (hunk 3 above)
Comment (anvitha-jain) [reply models/cloudtemplate_bgp_ipv4.go:25]: yes its in next line peer ASN

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**